src/components/hint/index.js

```diff
@@ -0,0 +1,335 @@
+    const {edgeSpace} = this.getThemeProps();
+    return edgeSpace || DEFAULT_EDGE_SPACE;
+  }
+
+  get useSideTip() {
+    const {useSideTip} = this.props;
+    if (!_.isUndefined(useSideTip)) {
+      return useSideTip;
+    }
+    return this.getTargetPositionOnScreen() !== 'center';
+  }
+
+  getTargetPositionOnScreen() {
+    const {targetLayout} = this.state;
+    const targetMidPosition = targetLayout.x + targetLayout.width / 2;
+    if (targetMidPosition > Constants.screenWidth * (2 / 3)) {
+      return 'right';
+    } else if (targetMidPosition < Constants.screenWidth * (1 / 3)) {
+      return 'left';
+    } else {
+      return 'center';
+    }
+  }
+
+  getContainerPosition() {
+    const {targetLayout} = this.state;
+    if (targetLayout) {
+      return {top: targetLayout.y, left: targetLayout.x};
+    }
+  }
+
+  getHintPosition() {
+    const {targetLayout} = this.state;
+    const {position} = this.props;
+    const hintPositionStyle = {left: -targetLayout.x, alignItems: 'center'};
+
+    if (position === HINT_POSITIONS.TOP) {
+      hintPositionStyle.bottom = 0;
+    } else {
+      hintPositionStyle.top = targetLayout.height;
+    }
+
+    const targetPositionOnScreen = this.getTargetPositionOnScreen();
+    if (targetPositionOnScreen === 'right') {
+      hintPositionStyle.alignItems = 'flex-end';
+    } else if (targetPositionOnScreen === 'left') {
+      hintPositionStyle.alignItems = 'flex-start';
+    }
+
+    return hintPositionStyle;
+  }
+
+  getHintPadding() {
+    const {targetLayout} = this.state;
+    const paddings = {paddingVertical: this.hintDistance, paddingHorizontal: this.edgeSpace};
+    if (this.useSideTip) {
+      const targetPositionOnScreen = this.getTargetPositionOnScreen();
+      if (targetPositionOnScreen === 'left') {
+        paddings.paddingLeft = targetLayout.x;
+      } else if (targetPositionOnScreen === 'right') {
+        paddings.paddingRight = Constants.screenWidth - targetLayout.x - targetLayout.width;
+      }
+    }
+
+    return paddings;
+  }
+
+  getTipPosition() {
+    const {targetLayout} = this.state;
+    const {position} = this.getThemeProps();
+    const tipPositionStyle = {};
+
+    if (position === HINT_POSITIONS.TOP) {
+      tipPositionStyle.bottom = this.hintDistance - this.tipSize.height;
+    } else {
+      tipPositionStyle.top = this.hintDistance - this.tipSize.height;
+    }
+
+    const targetMidWidth = targetLayout.width / 2;
+    const tipMidWidth = this.tipSize.width / 2;
+
+    switch (this.getTargetPositionOnScreen()) {
+      case 'left':
+        tipPositionStyle.left = this.useSideTip ? targetLayout.x : targetLayout.x + targetMidWidth - tipMidWidth;
+        break;
+      case 'right':
+        tipPositionStyle.right = this.useSideTip
+          ? Constants.screenWidth - targetLayout.x - targetLayout.width
+          : Constants.screenWidth - targetLayout.x - targetMidWidth - tipMidWidth;
+        break;
+      case 'center':
+      default: {
```

Comment:
why wrapping in an object here?

Reply:
it's not an object, it's just the case's scope curly brackets. 
Anyway, I'll remove it